pkg/controller/controller.go

```diff
@@ -89,25 +89,14 @@ func (c *Controller) Run() error {
 // CreateVolume implements pb.ControllerServer.CreateVolume
 func (c *Controller) CreateVolume(contx context.Context, opt *pb.CreateVolumeOpts) (*pb.GenericResponse, error) {
 	var err error
-	var prf *model.ProfileSpec
 	var snap *model.VolumeSnapshotSpec
 	var snapVol *model.VolumeSpec
 
 	log.Info("Controller server receive create volume request, vr =", opt)
 
 	ctx := osdsCtx.NewContextFromJson(opt.GetContext())
-	if opt.ProfileId == "" {
-		log.Warning("Use default profile when user doesn't specify profile.")
-		prf, err = db.C.GetDefaultProfile(ctx)
-		opt.ProfileId = prf.Id
-	} else {
-		prf, err = db.C.GetProfile(ctx, opt.ProfileId)
-	}
-	if err != nil {
-		db.UpdateVolumeStatus(ctx, db.C, opt.Id, model.VolumeError)
-		log.Error("get profile failed: ", err)
-		return pb.GenericResponseError(err), err
-	}
+	prf := model.NewProfileFromJson(opt.Profile)
```

Comment:
It should be ProfileId right?

Reply:
No we added `Profile` field in protobuf definition, so that southbound drivers can leverage whatever they want in Profile.

Reply:
But in this review commit change, I can't find protobuf file changes